tests/common.h

```diff
@@ -355,7 +355,7 @@ namespace witest
         TestFolder(const TestFolder&) = delete;
         TestFolder& operator=(const TestFolder&) = delete;
 
-        TestFolder(TestFolder&& other)
+        TestFolder(TestFolder&& other) WI_NOEXCEPT

```

Comment:
Did something really warn about a function _not_ being noexcept? I'm not arguing that this _shouldn't_ be noexcept, but I'm picturing a warning that says "function that does not throw should be declared as noexcept" which would be a terrible, terrible thing to have

Reply:
The warning was like "this kind of method may not throw" ... not "this method should be noexcept."  https://en.cppreference.com/w/cpp/language/move_constructor has:

> To make strong exception guarantee possible, user-defined move constructors should not throw exceptions. For example, std::vector relies on std::move_if_noexcept to choose between move and copy when the elements need to be relocated.

But I don't know how to compile the language spec in my head so I don't know if there's a hard requirement.

I'm happy to roll back changes in test code as the goal is to get the library itself analyze-clean.

Reply:
If it's only warning about move constructors/assignment operators, that's not as bad since it _is_ a desirable property to have such operations be `noexcept`. That said, if it gave the same warning for `wistd::function`, that's a pretty embarrassing bug in the analyzer. Either that or it's warning for _all_ move constructors, which would be worse and effectively renders the warning useless and something that should always be disabled.